src/renderer/versions.ts

```diff
@@ -179,6 +179,22 @@ export function addLocalVersion(input: Version): Array<Version> {
   return versions;
 }
 
+/**
+ * Check to see if local version has already been added
+ * for the current folder path
+ *
+ * @param {Version} input
+ * @returns {Version | undefined}
+ */
+export function getExistingLocalVersion(
+  folderPath: string,
+): Version | undefined {
+  const versions = getLocalVersions();
+  const duplicate = versions.find((v) => v.localPath === folderPath);
+
+  return duplicate ? (duplicate as Version) : undefined;
```

Comment:
Array.prototype.find() returns the item or undefined, so I think we should be able to return it as-is without having to check the results:

```suggestion
  return getLocalVersions().find((v) => v.localPath === folderPath);
```